packages/eslint-plugin/lib/rules/no-useless-constructor.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @fileoverview Disallow unnecessary constructors
+ * @author Armano <https://github.com/armano2>
+ */
+'use strict';
+
+const baseRule = require('eslint/lib/rules/no-useless-constructor');
+const util = require('../util');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = Object.assign({}, baseRule, {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow unnecessary constructors',
+      category: 'ECMAScript 6',
+      url: util.metaDocsUrl('no-useless-constructor')
+    },
+    schema: baseRule.meta.schema,
+    messages: baseRule.meta.messages
+  },
+
+  create(context) {
+    const rules = baseRule.create(context);
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+    return {
+      MethodDefinition(node) {
+        if (
+          !node.value ||
+          node.value.type !== 'FunctionExpression' ||
+          node.accessibility
```

Comment:
would it be good to check for not public?

i.e. this use constructor is still useless:
```TS
class X {
  public constructor() {}
}
```

Reply:
public one should be useless only when there is no super class?

Reply:
you are correct [repl](https://www.typescriptlang.org/play/index.html#src=class%20Parent%20%7B%0A%20%20%20%20protected%20constructor()%20%7B%20%7D%0A%7D%0A%0Aclass%20Child%20extends%20Parent%20%7B%20%7D%0A%0Aconst%20x%20%3D%20new%20Child()%3B%0A)

the public constructor def is only useless if there is no super class.

technically the public case is *not* useless if:
- there is a super class
- the super class defines a constructor
- the constructor has a different signature

But that's not checkable from within eslint without full type info

Reply:
ok, than i'm just going to check if there is super class and add test cases for that